.github/workflows/main.yaml

```diff
@@ -1,7 +1,7 @@
-# Built from:
-# https://docs.github.com/en/actions/guides/building-and-testing-python
+# Unified CI workflow for linkml monorepo
+# Tests both linkml and linkml-runtime packages together
```

Comment:
combining tests like this, imo, loses some of the advantages of separate packages within a monorepo, and has a pretty strong potential for making this and any other sub-packages non-independent (tests pass when both are installed together, but `linkml-runtime` can no longer be used by itself, accidentally). 

i think it would probably be good to have separate workflows for each of the sub-packages (and to reduce boilerplate one could make reusable local actions), and to use `path` filters so that, e.g. `linkml-runtime` tests are only run when `linkml-runtime` files are changed, and so on: https://docs.github.com/en/actions/reference/workflows-and-actions/workflow-syntax#onpushpull_requestpull_request_targetpathspaths-ignore